### PR TITLE
Fix Travis tests

### PR DESCRIPTION
**Pull request by vadimcn**
r? @vadimcn

### CODE REVIEW

**Comment by vadimcn** [comment]
cc #35011

**Comment by TimNN** [comment]
@vadimcn maybe better use a `println!` to prevent the double panic (and perhaps add some metadata, ie. the value of `version`, should that check fail, not sure which of the parameters are relevant)?

Also you could change [this line](https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs#L51) in compile test to a `println!` as well to hopefully know exactly which test is failing (as long as compiletest doesn't do stuff in parallel).

**Comment by vadimcn** [comment]
Can't println from panic_ unwind, as it precedes std in the dependency chain.  I was counting on panic! printing the line number.
In any case, I'm pretty sure it's the second return site.  We're probably trying to get a stack trace through some function that llvm has decided is nounwind.

**Comment by cgswords** [comment]
I'm also experiencing this interesting behavior in #37837

**Comment by alexcrichton** [comment]
@vadimcn FWIW I can reproduce locally inside the docker container (Ubuntu 16.04) and it appears to be the src/test/run-pass/backtrace.rs test failing which in turn causes compiletest to fail as well.

**Comment by vadimcn** [comment]
Mmkay... so we can fix this by reverting to `gcc_personality`'s behavior (i.e. allowing portions of code to be not covered by the unwind table).

**Comment by alexcrichton** [comment]
@vadimcn looks like tests passed (yay!), wanna merge this after cleaning up a bit?

**Comment by vadimcn** [comment]
Sure.  Though this may mask bugs like [this one](https://github.com/rust-lang/rust/pull/34832#issuecomment-233551113).  
r?

**Comment by alexcrichton** [comment]
Ah ok, maybe once we drop support for LLVM 3.7 we can try removing this clause?

@bors: r+ 5fbcf08dd8cfc347c13d84c8ea20e3bfce6793fc

**Comment by alexcrichton** [comment]
@bors: p=1 

(fixing travis)

**Comment by alexcrichton** [comment]
@bors: retry

* good ol network failures

**Comment by steveklabnik** [comment]
welp

File: src/libpanic_unwind/dwarf/eh.rs [hunk 1]
```diff
@@ -108,10 +108,9 @@ pub unsafe fn find_eh_action(lsda: *const u8, context: &EHContext) -> EHAction {
                 }
             }
         }
-        // If ip is not present in the table, call terminate.  This is for
-        // a destructor inside a cleanup, or a library routine the compiler
-        // was not expecting to throw
-        EHAction::Terminate
+        // Ip is not present in the table.  This should not hapen... but it does: issie #35011.
```
Comment (jseyfried) [inline src/libpanic_unwind/dwarf/eh.rs]: typo: *happen*, *issue*

**Comment by alexcrichton** [comment]
@bors: retry

* https://github.com/rust-lang/rust/pull/34983

**Pull request merged**